script.js

```diff
@@ -1,4 +1,165 @@
+var GAME_STATE_ROLL_DICE = "GAME_STATE_ROLL_DICE ";
+var GAME_STATE_CHOOSE_DICE_ORDER = "GAME_STATE_CHOOSE_DICE_ORDER";
+var GAME_STATE_COMPARE_SCORES = "GAME_STATE_COMPARE_SCORES";
+var GAME_STATE_PLAY_AGAIN = "GAME_STATE_PLAY_AGAIN";
+var gameState = GAME_STATE_ROLL_DICE;
+var currentPlayer = 1;
+var currentPlayerRolls = []; //this array will store the currrent player rolls
+var allPlayersScore = [];
+var playerFinalValue = 0;
+var errorMessage = "";
+//Helper function "Roll the dice"
+var rollDice = function () {
+  var randomDecimal = Math.random() * 6;
+  var randomInteger = Math.floor(randomDecimal);
+  var diceRoll = randomInteger + 1;
+  return diceRoll;
+};
+
+//Helper function "Roll the dice x2"
+var rollDiceForPlayer = function () {
+  var counter = 0;
+  while (counter < 2) {
+    currentPlayerRolls.push(rollDice()); // adding dice rolls to the player rolls array on line 16
+    counter = counter + 1;
+  }
+
+  return (
+    "Welcome Player " +
+    currentPlayer +
+    ". <br> <br> You rolled " +
+    currentPlayerRolls[0] +
+    " and " +
+    currentPlayerRolls[1] +
+    ".<br> <br> Now please key in 1 or 2 to choose the corresponding dice as the first digit of your final value."
+  );
+};
+//helper function to get the player final value
+var getPlayerFinalValue = function (input) {
+  //input validation:
+  if (input != 1 && input != 2) {
+    errorMessage =
+      "Error! Please only key in 1 or 2 to choose which dice to use as the first digit. Your current dice rolls are " +
+      currentPlayerRolls[0] +
+      " as dice 1 and " +
+      currentPlayerRolls[1] +
+      " as dice 2.";
+    return errorMessage;
+  } else {
+    //condition 1 "if input == 1"
+    if (input == 1) {
+      playerFinalValue = Number(
+        `${currentPlayerRolls[0]}` + `${currentPlayerRolls[1]}`
+      );
+    }
+
+    //condition 2 "if input == 2"
+    if (input == 2) {
+      playerFinalValue = Number(
+        `${currentPlayerRolls[1]}` + `${currentPlayerRolls[0]}`
+      );
+    }
+    allPlayersScore.push(playerFinalValue); // add the final values to the corresponding array on line 19
+    currentPlayerRolls = []; //clear the current player rolls array
+
+    return (
+      "Player " +
+      currentPlayer +
+      ", your final value is " +
+      playerFinalValue +
+      "."
+    );
+  }
+};
+var comparePlayersScore = function () {
+  //player 1 wins:
+  if (allPlayersScore[0] > allPlayersScore[1]) {
+    return (
+      "Player's 1 final value is " +
+      allPlayersScore[0] +
+      ". <br> <br> Player's 2 final value is " +
+      allPlayersScore[1] +
+      ". <br> <br> " +
+      "Player 1 wins."
+    );
+    //player 2 wins
+  } else if (allPlayersScore[1] > allPlayersScore[0]) {
+    return (
+      "Player's 1 final value is " +
+      allPlayersScore[0] +
+      ". <br> <br> Player's 2 final value is " +
+      allPlayersScore[1] +
+      ". <br> <br> " +
+      "Player 2 wins."
+    );
+    // a tie:
+  } else {
```

Comment:
`else` is not necessary, whatever remains here must be `allPlayersScore[0] === allPlayersScore[1]` implicitly